### ISSUE TITLE
Allow ability for users to search by distance

### ISSUE THREAD

**Issue by pH-7** [pH7Software/pH7-Social-Dating-CMS#537]
On user search form https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/master/_protected/app/system/core/forms/SearchUserCoreForm.php

- Allows them to select a distance radius (kilometers or miles for the remaining countries using imperial unit).
- Add geocoding (longitude/latitude, and radius) to detect user's coordinates.

**Comment by rondlite** [comment]
use geohash or other like https://github.com/ashwin711/proximityhash
Redis is quite good at handling hash locations. 
Lat long will require a lot more logic than geohash..

**Comment by rondlite** [comment]
Added the search by distance on my site, without adding yet another GeoAPI
Pretty easy when adding location of user on register and store them in Redis too.

**Comment by Undefined-Variables** [comment]
Yeah I have written a how far feature that tells you how far apart based on postcodes and also gives time by car/train/bus and the cost of each. This is all doable using googles distance matrix :)

**Comment by rondlite** [comment]
@Undefined-Variables with redis you don't need google api .. will save you some money

**Comment by polynamaude** [comment]
@rondlite It's rightt, you only have to save lon / lat and make the calculation...
Great hint "Ron-D-Lite"... (Don't get confused with Rob Da Bank, that's another famous star of the night)

**Comment by rondlite** [comment]
> Don't get confused with Rob Da Bank, 
he is Lazyboy (serious looooooool) (good stuff)

**Comment by Undefined-Variables** [comment]
Yeah I was given £500 google credit and its using about £6 a month so not costing me a penny... 

@rondlite are you calling me lazy? If that is the light you see me in you have gotten me completely wrong!!!!

**Comment by polynamaude** [comment]
@Undefined-Variables I think you shall put your head in the ice or whatever is suitable to cool down your mind a bit....
Ron-D-Lite is Ron's nick name as both as DJ and light technician in the Netherlands club scene. Rob-da-Bank is another DJ so is Lay-Z-Boy...

Come down a bit...
To be so much easily pissed off ain't much british...

**Comment by Undefined-Variables** [comment]
LOL I'm not pissed off I didn't understand is all. You don't seriously think I get pissed of that easy do you? In britain we are taught to ask the question if you are not sure... We was talking about Google and the pricing, then he said "Hes just lazy" I just assumed he was still on the same subject...

TBH I like @rondlite and if anything was concerned I look lazy haha! How are you @polynamaude any how?

**Comment by Undefined-Variables** [comment]
@polynamaude you have a magical way with words. Word smith lol...

@rondlite thanks for tip :) sorry I misunderstood the off topic DJ references...

**Comment by rondlite** [comment]
@Undefined-Variables the mention of Rob da Bank made me quote it is lazyboy.. not calling anyone lazy.
```
Birth name | Robert John Gorham
-- | --
Also known as: Rob da Bank
24 June 1973 (age 47)Portsmouth, Hampshire, England, UK
Associated acts: Lazyboy
```
I am calling Rob da Bank Lazyboy, since he is. It is his act.  
He is also presenter of the Blue Room on BBC Radio 1.

**Comment by polynamaude** [comment]
@rondlite Hi Ron, glad to see ya alive ! I know that this virus ain't gonna kill you...

**Comment by Undefined-Variables** [comment]
@polynamaude also alive... Sorry been so busy lately!

**Comment by pH-7** [comment]
Thanks for the suggestion.

I reviewed this during backlog cleanup. This request is valid but outside the current 18.x delivery scope, so I’m closing it as not planned for now to keep the tracker focused on active work.

If you want to push this forward, please open a new proposal with a minimal implementation plan (scope, DB/API impact, migration notes, and backward-compatibility expectations).

Take care and thank you for using pH7 Social Dating Builder,

[Pierre-Henry](https://github.com/pH-7)
